### PR TITLE
BookA-Pset_ChemicalLabTestSolid

### PR DESCRIPTION
**Pull request by larswik**
# Pset: Pset_ChemicalLabtestSolid

## Description: 

Properties for results of lab tests for chemical solids.

| Property | Datatype | Description |
|----------|----------|-------------|
|Zn|IfcReal|content of Zn in the water sample.|
|Cu|IfcReal|content of Cu in the water sample.|
|Pb|IfcReal|content of Pb in the water sample.|
|Cr|IfcReal|content of Cr in the water sample.|
|Hg|IfcReal|content of Hg in the water sample.|
|Ni|IfcReal|content of Ni in the water sample.|
|As|IfcReal|content of As in the water sample.|
|Cd|IfcReal|content of Cd in the water sample.|
|Mn|IfcReal|content of Mn in the water sample.|
|OrganicMatterContent|IfcReal|The total carbon content in soil (test method after dry combustion, the organic carbon content is calculated from this content after correcting for carbonates present in the sample).|
|TS|IfcReal|The total sulfur content.|

### CODE REVIEW

**Comment by patmclarin** [comment]
Please update description to: Properties for results of lab tests for chemical solids

**Review by SergejMuhic — COMMENTED** [review]
Also propose a meeting as for the lab test liquid.

**Comment by larswik** [comment]
> Also propose a meeting as for the lab test liquid.

Replaced with Pset_MaterialSolid (same principle as Pset_MaterialWater). A bit unclear about the units. Some speak about mg/L and others about ppm or mg/kg. This must be reviewed.

File: IFC4x3/Properties/t/TS_3TIrUqMFzEHxl56bjVXyxc/DocProperty.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="TS_3TIrUqMFzEHxl56bjVXyxc" Name="TS" UniqueId="dd4b57b4-58ff-4e47-bbc5-1a5b5f87cee6" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/t/TS_3TIrUqMFzEHxl56bjVXyxc/DocProperty.xml:2]: What does TS mean?

File: IFC4x3/Sections/Resource definition data schemas/Schemas/IfcMaterialResource/PropertySets/Pset_MaterialSolid/Documentation.md [hunk 2]
```diff
@@ -0,0 +1 @@
+A set of extended material properties for of solid (natural) materials typically used within the context of geology and geotechnics.
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Resource definition data schemas/Schemas/IfcMaterialResource/PropertySets/Pset_MaterialSolid/Documentation.md]:
```suggestion
A set of extended material properties of solid (natural) materials typically used within the context of geology and geotechnics.
```

**Review by SergejMuhic — CHANGES_REQUESTED** [review]
I suggest adding "The unit must be defined in IfcPropertySingleValue.Unit." to all "XContent" properties and deriving an issue for a "Single Value with Unit" concept template that will allow us to require IfcPropertySingleValue.Unit on these properties.

File: IFC4x3/Properties/t/TS_3TIrUqMFzEHxl56bjVXyxc/DocProperty.xml (hunk 1 above)
Comment (larswik) [reply IFC4x3/Properties/t/TS_3TIrUqMFzEHxl56bjVXyxc/DocProperty.xml:2]: It says "The total sulfur content" in the description.

**Pull request merged**